test/bootstrap/checkout-pr.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+####################################################
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+####################################################
+
+set -eu -o pipefail
+set -x
+
+git init .
+git clean -dfx
+git reset --hard
+git config --local user.name 'ACS Bot'
+git config --local user.email 'acs-bot@microsoft.com'
+git fetch --tags https://github.com/${REPO_OWNER}/${REPO_NAME} master +refs/pull/${PULL_NUMBER}/head:refs/pr/${PULL_NUMBER}
+git checkout -B test "${PULL_BASE_SHA}"
```

Comment:
They get separate workspaces. (In fact, in the system I showed you, each build job happens in a new Kubernetes Pod - clean slate everytime, etc).